src/components/shared/form_inputs/FunDynamicSubmitButton.vue

```diff
@@ -0,0 +1,37 @@
+<template>
+	<button
+		class="button"
+		:type="buttonType"
+		v-bind="$attrs"
+	>
+		{{ $t( dynamicButtonText ) }}
+	</button>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue';
+
+interface Props {
+	paymentType: 'PPL' | 'UEB' | 'MCP' | 'SUB';
+	buttonType?: 'button'|'submit'|'reset';
+}
+
+const props = withDefaults( defineProps<Props>(), {
+	buttonType: 'button',
+} );
+
+const dynamicButtonText = computed( () => {
+	switch ( props.paymentType ) {
+		case 'PPL':
+			return 'donation_form_finalize_paypal';
+		case 'UEB':
+			return 'donation_form_finalize_bank_transfer';
+		case 'MCP':
+			return 'donation_form_finalize_credit_card';
+		case 'SUB':
+			return 'donation_form_finalize_sofort';
+		default:
+			return 'donation_form_finalize';
```

Comment:
The dynamic button text will lead to problems and extra effort (to avoid false positives) when we do the "unused key checker" script ( https://phabricator.wikimedia.org/T346733 ). I suggest using the translation function inside `dynamicButttonText` instead of inside the template, that'll make the "search for calls to the translation function" of the key checker script easier.
See https://stackoverflow.com/a/69103578/130121 for an example on how to use VueI18N inside the script (setup) section of a Vue component